### PR TITLE
Gke build (WIP)

### PR DESCRIPTION
**Pull request by anandkumarpatel**

### CODE REVIEW

File: lib/workers/docker.event.publish.js [hunk 1]
```diff
@@ -48,41 +48,107 @@ const inspectSchema = Joi.object({
...
+DockerEventPublish._inspectImage = (job, newEvent) => {
+  log.trace('DockerEventPublish _inspectImage')
+  const docker = new Docker(job.Host, job.org)
+
+  if (!DockerEventPublish._isBuildContainerDeath(job)) {
+    return newEvent
+  }
+
+  return Promise.fromCallback((cb) => {
+    docker.getImage(job.Actor.Attributes.dockerTag).inspect(cb)
```
Comment (podviaznikov) [inline lib/workers/docker.event.publish.js:129]: I don't think `Actor` and `Attributes` are required

File: test/unit/worker/docker.event.publish.js [hunk 2]
```diff
@@ -89,9 +95,32 @@ describe('docker event publish', () => {
...
+
+    it('should call image inspect when needsInspect true', (done) => {
+      const testJob = eventMock({
+        needsInspect: true,
+        status: 'die'
+      })
+      const testInspect = { Id: 'gear' }
+
+      keypather.set(testJob, 'Actor.Attributes.type', 'image-builder-container')
+      DockerClient.prototype.inspectContainerAsync.returns(Promise.resolve())
```
Comment (podviaznikov) [inline test/unit/worker/docker.event.publish.js]: you can import `sinon-as-promised` and just do `DockerClient.prototype.inspectContainerAsync.resolves()`

**Review by podviaznikov — CHANGES_REQUESTED** [review]

File: lib/workers/docker.event.publish.js [hunk 3]
```diff
@@ -48,41 +48,107 @@ const inspectSchema = Joi.object({
   })
 }).unknown()
 
+const imageInspectSchema = Joi.object({
+  Config: Joi.object({
+    Cmd: Joi.alternatives().try(Joi.array().items(Joi.string()), null),
```
Comment (podviaznikov) [inline lib/workers/docker.event.publish.js:53]: can't you just say `Joi.array().items(Joi.string())`? is it not the same as alternative between something and `null`?

File: lib/workers/docker.event.publish.js (hunk 1 above)
Comment (anandkumarpatel) [reply lib/workers/docker.event.publish.js:129]: they are based on type

File: test/unit/worker/docker.event.publish.js (hunk 2 above)
Comment (anandkumarpatel) [reply test/unit/worker/docker.event.publish.js]: done

**Review by podviaznikov — APPROVED** [review]

File: lib/workers/docker.event.publish.js (hunk 3 above)
Comment (anandkumarpatel) [reply lib/workers/docker.event.publish.js:53]: no, sadly null is a value.

**Pull request merged**